kernel/src/kernel_predicates/mod.rs

```diff
@@ -210,28 +210,27 @@ pub(crate) trait KernelPredicateEvaluator {
         use BinaryPredicateOp::*;
         use Expr::{Column, Literal};
 
-        // NOTE: We rely on the literal values to provide logical type hints. That means we cannot
-        // perform column-column comparisons, because we cannot infer the logical type to use.
-        let (op, col, val) = match (left, right) {
-            (Column(a), Column(b)) => return self.eval_pred_binary_columns(op, a, b, inverted),
-            (Literal(a), Literal(b)) => return self.eval_pred_binary_scalars(op, a, b, inverted),
-            (Literal(val), Column(col)) => (op.commute()?, col, val),
-            (Column(col), Literal(val)) => (op, col, val),
+        match (left, right) {
+            (Column(a), Column(b)) => self.eval_pred_binary_columns(op, a, b, inverted),
+            (Literal(a), Literal(b)) => self.eval_pred_binary_scalars(op, a, b, inverted),
+            (Column(col), Literal(val)) => match op {
+                LessThan => self.eval_pred_lt(col, val, inverted),
+                GreaterThan => self.eval_pred_gt(col, val, inverted),
+                Equal => self.eval_pred_eq(col, val, inverted),
+                Distinct => self.eval_pred_distinct(col, val, inverted),
+                In => self.eval_pred_in(col, val, inverted),
+            },
+            (Literal(val), Column(col)) => match op {
```

Comment:
Could you add a comment that this is the "swapped" order (and that's why we do a `gt` for a `LessThan`. It was sort of clear before due to the `commute`, but it's a bit confusing now until you notice.